### DISCUSSION TITLE
[Question / Suggestion] How do rockets land?

### DISCUSSION THREAD

**Discussion by cloudrambler** [tsunamayo/Starship-EVO#4032] [Suggestions]
There are two logical orientation formats for space ships; plane style and rocket style. The plane style has a forward vector that is distinct from its up vector (this is what we all seem to be building). The rocket style would logically have the same vector for forward and up; the the game presently forces them distinct.

This is not a problem for ship building as we can define the effect of these vectors by orienting the helm yoke. Likewise we can configure our deck orientation with the gravity units. Plus I understand for flight control that there need to be three distinct control vectors to map the controls to.

But what about landing on the shipyard? The landing process aligns the ship's up vector with the landing pad's up vector. This would force a rocket style ship to land on its side rather than its feet.

Another example is the "Slave One" ship from Star Wars, as that flies forward and lands on its back. For that ship to land correctly its default up vector could not be aligned to the shipyard's up vector.

We could change the vectors for a special shipyard, but for interchangeable (common use) shipyards, there is a compatibility problem. Plus for the shipyard to work properly, its up vector should be aligned normal to the surface of the landing pad.

My suggestion is that there be a configuration option within the ship computer, allowing "landing vertical" to be set as a distinct vector. A matrix value method as per the shipyard and gravity devices would allow for non-oblique designs and could be more universal than a toggle option.

**Comment by StarrKiss** [comment]
Completely agree- built TinTins moon rocket a while ago and had this issue.

**Comment by luckyrada** [comment]
what if you build the ship upward?

**Comment by tsunamayo** [comment]
Yep this is a valid point, I will add a setting in the shipcore. Thanks!

**Comment by cloudrambler** [comment]
@luckyrada The flight and landing vectors are determined by the helm yoke orientation. It would seem that these helm vectors overrule the default grid vectors that are generated when you place the starting block. So changing the build orientation does not add further control of the problem.

If you build a rocket style ship with the up vector aligned with the main axis, so that it lands correctly, then its forward vector and flight direction would have to be horizontal. It would fly sideways. This would look silly, but more importantly, you would get zero thrust or lightcruise power from your main thrusters.
